src/Datagrid/ProxyQueryInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineMongoDBAdminBundle\Datagrid;
+
+use Doctrine\ODM\MongoDB\Query\Builder;
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface as BaseProxyQueryInterface;
+
+interface ProxyQueryInterface extends BaseProxyQueryInterface
+{
+    /**
+     * @return Builder
+     */
+    public function getQueryBuilder();
```

Comment:
```suggestion
    public function getQueryBuilder(): Builder;
```

?

Reply:
Plus remove the phpdoc then

Reply:
This cannot be done on 3.x. Since the ProxyQuery provided does not have return type currently.

Reply:
True :+1: